core/src/main/scala/caliban/introspection/Introspector.scala

```diff
@@ -44,8 +63,15 @@ object Introspector {
       ),
       args => types.find(_.name.contains(args.name)).get
     )
-    val introspectionSchema = Schema.gen[__Introspection]
-    RootSchema(Operation(introspectionSchema.toType_(), introspectionSchema.resolve(resolver)), None, None)
+
+    for {
+      wrappedInto <- wrap(ZIO.succeed(resolver))(introWrappers)
```

Comment:
If you move this line to inside `RootSchema`, like `QueryStep(ZQuery.fromEffect(wrap(ZIO.succeed(resolver))(introWrappers).flatMap(wrappedInto => introspectionSchema.resolve(wrappedInto))`, then `introspect` would not need to return an effect. Unless I missed something, I think it can work?

Reply:
seems you were right, fixed

Reply:
@ghostdogpr

Reply:
It looks good 👍
I didn't have the time to merge and cut a new release this weekend, I'll do it some time this week.